### ISSUE TITLE
[Bug]: Ram issue: just four running job, then 4G ram used..

### ISSUE THREAD

**Issue by meisuj815** [openbullet/OpenBullet2#454]
### Version of the software

0.1.23

### Operating system

Debian 10

### Browser / Native

Firefox

### What happened?

[image of Ram usage](https://img.onl/dacBPc)

just four running job, then 4G ram used..
Hi, which tools should I use for dump the memory for you to debug?

for privacy season not showing code in here, if you want the LoliCode ask me again

thanks alot for help

**Comment by openbullet** [comment]
Please use `dotnet dump` as shown in this blog post, then simply send me the dump file (you don't need to analyze it yourself)

**Comment by meisuj815** [comment]
Hi @openbullet 

Thanks for reply. I already dump the ram via this command:

```
apt -y install dotnet-sdk-5.0
dotnet tool install --global dotnet-dump
dotnet dump collect -p 3932
```

The file had 3.7G big.. I will zip the file and upload to google drive for you to debug, please wait a while.

**Comment by meisuj815** [comment]
https://file.io/0xkyqO0kNg0V

Wow, after 7z the file only had 84MB

**Comment by openbullet** [comment]
Yes because there is a lot of repeated data and compression is really good at handling that. Anyways I will take a look at it and let you know.

**Comment by bestisben-xyz** [comment]
![image](https://user-images.githubusercontent.com/90732027/133650342-305950bf-eb0c-4aca-bb20-f05cf65f49b2.png)

now the RAM almost running out..

**Comment by openbullet** [comment]
I am going to look into it, please understand that fixing things can take time since this is now my full time job and I have other things to do and other issues to prioritize ^^

**Comment by bestisben-xyz** [comment]
Hi @openbullet 

Sure bro, that is not a problem, I am totally understand. I am very thankful for what you did:)

**Comment by shakshak12** [comment]
![image](https://user-images.githubusercontent.com/87039966/158345692-9cdbf665-995b-4464-ac03-ce6af4286636.png)
Hey same issue is here no job is running high Ram usage (4 jobs already finished with huge custom)

![image](https://user-images.githubusercontent.com/87039966/158345830-ecf6fd94-5105-446d-9b3f-188d72afae94.png)

**Comment by openbullet** [comment]
Hey @shakshak12 the high ram usage is probably due to the fact that the jobs that finished still contain some information about the data/proxies/hits inside them. That information usually gets deleted when you run jobs again. Please make sure the Bot Log is not enabled in RL settings, because that is the main source of ram usage in the entire program.

**Comment by shakshak12** [comment]
Bot log is not enabled and the issue is still there

**Comment by openbullet** [comment]
Okay, please take a memory dump as instructed on issue https://github.com/openbullet/OpenBullet2/issues/454
I will close this since it's really similar, we will continue discussing there

**Comment by openbullet** [comment]
Please take a memory dump as instructed in a few posts above and send it to me, thanks!

**Comment by shakshak12** [comment]
Thanks for responding but sorry I am not able to dump the memory, I am running OB2 using docker and PID is different each time 
![image](https://user-images.githubusercontent.com/87039966/159141762-421d9454-7c9c-4c38-8eba-55bd79ccf058.png)
![image](https://user-images.githubusercontent.com/87039966/159141815-d3ec4513-fbf6-4de7-b990-b7719dcc2134.png)

**Comment by shakshak12** [comment]
trying with ps aux to get the Process ID 
![image](https://user-images.githubusercontent.com/87039966/159161254-2103c690-4de3-492a-9c12-5d734dc7b161.png)

trying this command 

![image](https://user-images.githubusercontent.com/87039966/159161351-36000a8d-78e2-4bba-a315-6e8f2492fdeb.png)

**Comment by JReming85** [comment]
You can do it within the container

docker ps to get the container id if you dont know the name
docker exec <container id or name> <cmd>
